torch_xla/torch_xla.py

```diff
@@ -83,9 +84,12 @@ def step():
   return compile()
 
 
-def compile(f: Optional[Callable] = None,
-            full_graph: Optional[bool] = False,
-            name: Optional[str] = None):
+def compile(
+    f: Optional[Callable] = None,
+    full_graph: Optional[bool] = False,
+    name: Optional[str] = None,
+    detect_dynamic_shape=False,
```

Comment:
I think instead of a boolean value `detect_dynamic_shape`, we can have it to be something like `max_dynamic_shape_graph_allowed`, then you can map it to `_dynamic_shape_detector_set_max_allowed_traces ` directly.

Reply:
What do you think about `allowed_traces` (because we can have multiple graphs when tracing a function once)?